homeassistant/components/statistics/sensor.py

```diff
@@ -357,15 +358,14 @@ def __init__(
         self.samples_keep_last: bool = samples_keep_last
         self._precision: int = precision
         self._percentile: int = percentile
-        self._value: StateType | datetime = None
-        self._unit_of_measurement: str | None = None
+        self._value: float | int | datetime | None = None
```

Comment:
Still not a fan of this change. Isn't `StateType` provided by HA for a reason!? @emontnemery

Reply:
`StateType` is the union of the types a state is allowed to have. If `self._value` can't possible be a `str`, I think it's  clearer to be explicit.